### ISSUE TITLE
Unsatisfiable requirements detected for package Compat [34da2185]

### ISSUE THREAD

**Issue by fedario** [TuringLang/DistributionsAD.jl#46]
When I try to install the latest version of DistributionsAD, I get the following error:

```
(v1.3) pkg> add DistributionsAD@0.4.4
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package Compat [34da2185]:
 Compat [34da2185] log:
 ├─possible versions are: [1.0.0-1.0.1, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.5.0-1.5.1, 2.0.0, 2.1.0, 2.2.0, 3.0.0, 3.1.0, 3.2.0, 3.3.0-3.3.1, 3.4.0, 3.5.0, 3.6.0, 3.7.0, 3.8.0] or uninstalled
 ├─restricted by compatibility requirements with Weave [44d3d7a6] to versions: [1.0.0-1.0.1, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.5.0-1.5.1, 2.0.0, 2.1.0, 2.2.0, 3.0.0, 3.1.0, 3.2.0, 3.3.0-3.3.1, 3.4.0, 3.5.0, 3.6.0, 3.7.0, 3.8.0]
 │ └─Weave [44d3d7a6] log:
 │   ├─possible versions are: [0.6.0-0.6.2, 0.7.0-0.7.2, 0.8.0-0.8.1, 0.9.0-0.9.3] or uninstalled
 │   └─restricted to versions 0.9.3 by an explicit requirement, leaving only versions 0.9.3
 ├─restricted by compatibility requirements with DataFrames [a93c6f00] to versions: [2.0.0, 2.1.0, 2.2.0, 3.0.0, 3.1.0, 3.2.0, 3.3.0-3.3.1, 3.4.0, 3.5.0, 3.6.0, 3.7.0, 3.8.0]
 │ └─DataFrames [a93c6f00] log:
 │   ├─possible versions are: [0.11.7, 0.12.0, 0.13.0-0.13.1, 0.14.0-0.14.1, 0.15.0-0.15.2, 0.16.0, 0.17.0-0.17.1, 0.18.0-0.18.4, 0.19.0-0.19.4, 0.20.0-0.20.2] or uninstalled
 │   └─restricted to versions 0.20.2 by an explicit requirement, leaving only versions 0.20.2
 ├─restricted by compatibility requirements with DistributionsAD [ced4e74d] to versions: [3.6.0, 3.7.0, 3.8.0]
 │ └─DistributionsAD [ced4e74d] log:
 │   ├─possible versions are: [0.1.0-0.1.4, 0.2.0, 0.3.0-0.3.2, 0.4.0-0.4.4] or uninstalled
 │   └─restricted to versions 0.4.4 by an explicit requirement, leaving only versions 0.4.4
 └─restricted by compatibility requirements with YAML [ddb6d928] to versions: [1.0.0-1.0.1, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.5.0-1.5.1, 2.0.0, 2.1.0, 2.2.0] — no versions left
   └─YAML [ddb6d928] log:
     ├─possible versions are: 0.3.0-0.3.2 or uninstalled
     └─restricted by compatibility requirements with Weave [44d3d7a6] to versions: 0.3.0-0.3.2
       └─Weave [44d3d7a6] log: see above
```

I'm not sure what the issue is but it seems like a dependency problem.
Here is my pkg status:
```
(v1.3) pkg> status
    Status `~/.julia/environments/v1.3/Project.toml`
  [1520ce14] AbstractTrees v0.3.2
  [76274a88] Bijectors v0.5.2
  [9e28174c] BinDeps v1.0.0
  [336ed68f] CSV v0.5.26
  [3895d2a7] CUDAapi v3.1.0
  [3a865a2d] CuArrays v1.7.3
  [a93c6f00] DataFrames v0.20.2
  [31a5f54b] Debugger v0.6.4
  [31c24e10] Distributions v0.22.5
  [ced4e74d] DistributionsAD v0.3.2
  [89b67f3b] ExcelFiles v1.0.0
  [587475ba] Flux v0.10.3 #master (https://github.com/FluxML/Flux.jl.git)
  [bd48cda9] GraphRecipes v0.5.1
  [4c0ca9eb] Gtk v1.1.3
  [708ec375] Gumbo v0.7.0
  [cd3eb016] HTTP v0.8.12
  [7073ff75] IJulia v1.21.1
  [8197267c] IntervalSets v0.3.2
  [c8e1da08] IterTools v1.3.0
  [682c06a0] JSON v0.21.0
  [093fc24a] LightGraphs v1.3.1
  [cc2ba9b6] MLDataUtils v0.5.0
  [6f286f6a] MultivariateStats v0.7.0
  [47be7bcc] ORCA v0.3.1
  [5fb14364] OhMyREPL v0.5.5
  [90014a1f] PDMats v0.9.12
  [9b87118b] PackageCompiler v1.1.0
  [ccf2f8ad] PlotThemes v1.0.1
  [f0f68f2c] PlotlyJS v0.13.1
  [91a5bcdd] Plots v0.29.7
  [c46f51b8] ProfileView v0.6.3
  [438e738f] PyCall v1.91.4
  [d330b81b] PyPlot v2.8.2
  [e4f92426] ROC v0.1.0 #master (https://github.com/diegozea/ROC.jl)
  [b8865327] UnicodePlots v1.2.0
  [44d3d7a6] Weave v0.9.3
  [796a5d58] WordTokenizers v0.5.4
  [fdbf4ff8] XLSX v0.6.1
```

**Comment by devmotion** [comment]
YAML, a dependency of Weave, is not compatible with Compat 3.6 which is required by DistributionsAD. There's already some discussion about releasing an updated version of YAML (the last one was released 15 months ago apparently): https://github.com/BioJulia/YAML.jl/issues/73

For the time being, I guess you could solve the issue by removing Weave or working with separate [package environments](https://julialang.github.io/Pkg.jl/v1/environments/).

**Comment by SebastianM-C** [comment]
https://github.com/BioJulia/YAML.jl/pull/76 has been merged and YAML no longer depends on Compat. In the meantime Compat 3.8 was released (see #48).

**Comment by mohamed82008** [comment]
Closing this as it seems to be fixed in YAML.
